Community flashcards

winstonc03 opened the pull request:
Implemented Search use case with GUI.

lucasprates74 reviewed (commented):
Looks good! A couple notes though:
If the action methods for SearchScreen are defined in the constructor, the screen does not need to implement action listener

For the study and test cases, it might be better to create a new controller in the action listener for each "study" and "search" button. At least for the study mode i know that the use case interactor contains mutable data, and there is nothing in our GUI preventing the user from activating two study modes at once. This could lead to buggy behaviors. Maybe instead of passing the controllers into the SearchScreen and the ResultsScreen, just pass DBGateway and use that to create new controllers inside the action listener. I'd be happy to get someone else's opinion on this though. 

 Also maybe the interactor should check if the tag contains the input, not equals it. That way you wouldn't have to search the exact description of a flashcard set to find it for example. If this is difficult, we can change it after the first submission. I don't think its a big issue and shouldn't be a priority.

winstonc03 commented:
> Looks good! A couple notes though: If the action methods for SearchScreen are defined in the constructor, the screen does not need to implement action listener
> 
> For the study and test cases, it might be better to create a new controller in the action listener for each "study" and "search" button. At least for the study mode i know that the use case interactor contains mutable data, and there is nothing in our GUI preventing the user from activating two study modes at once. This could lead to buggy behaviors. Maybe instead of passing the controllers into the SearchScreen and the ResultsScreen, just pass DBGateway and use that to create new controllers inside the action listener. I'd be happy to get someone else's opinion on this though.
> 
> Also maybe the interactor should check if the tag contains the input, not equals it. That way you wouldn't have to search the exact description of a flashcard set to find it for example. If this is difficult, we can change it after the first submission. I don't think its a big issue and shouldn't be a priority.

Thanks for the feedback Lucas, I added in both of your recommendations in my recent commit. The Interactor now passes in the gateway instead of the two controllers, and the controllers are created in the action listeners in the Result Screen. I also changed the search results to check if the FlashcardSets **contain** the input rather than **equal** so it's easier for the user to search as well.

lucasprates74 reviewed (approved):
Looks great- I think its ready to merge!
